src/main/frontend/context/i18n.cljs

```diff
@@ -1,29 +1,27 @@
 (ns frontend.context.i18n
+  "Handles translation for the entire application. The dependencies for this ns
+  must be small since it is used throughout the application."
   (:require [frontend.dicts :as dicts]
-            [frontend.modules.shortcut.dict :as shortcut-dict]
-            [medley.core :refer [deep-merge]]
+            [frontend.modules.shortcut.dicts :as shortcut-dicts]
+            [tongue.core :as tongue]
             [frontend.state :as state]))
 
-;; TODO
-;; - [x] Get the preferred language from state
-;; - [x] Update the preferred language
-;; - [x] Create t functiona which takes a keyword and returns text with the current preferred language
-;; - [x] Add fetch for local browser preferred language if user has set it already
-;; - [ ] Fetch preferred language from backend if user is logged in
+(def dicts
+  (merge-with merge dicts/dicts shortcut-dicts/dicts))
 
-(defn fetch-local-language []
-  (.. js/window -navigator -language))
-
-(defonce translate-dicts (atom {}))
+(def translate
```

Comment:
This was unnecessarily stateful. We know of all the translations we need at compile time. No need to index by lang as tongue already handles this internally. See https://github.com/tonsky/tongue#usage for examples